setuptools/tests/test_build_meta.py

```diff
@@ -440,18 +442,28 @@ def run():
     def test_sys_argv_passthrough(self, tmpdir_cwd):
         path.build(self._sys_argv_0_passthrough)
         build_backend = self.get_build_backend()
-        with pytest.raises(AssertionError):
-            build_backend.build_sdist("temp")
+
+        # TODO: Clarify the use case motivating this test
+        # Why is it necessary that the assertion error is raised?
+        # Or is it just a side effect that was found to happen,
+        # but it is not necessary for setuptools to work properly?
+        # >>> with pytest.raises(AssertionError):
+
+        build_backend.build_sdist("temp")
 
     @pytest.mark.parametrize('build_hook', ('build_sdist', 'build_wheel'))
     def test_build_with_empty_setuppy(self, build_backend, build_hook):
         files = {'setup.py': ''}
         path.build(files)
 
-        with pytest.raises(
-                ValueError,
-                match=re.escape('No distribution was found.')):
-            getattr(build_backend, build_hook)("temp")
+        # TODO: Clarify why is it necessary that the error is raised?
```

Comment:
Maybe it's related to an empty `setup.py` with no metadata (not even a package name)?

Reply:
The solution for this is presented in  https://github.com/pypa/setuptools/commit/660a273f67dc0aa3e9d2818079051a4b927d91bf (https://github.com/pypa/setuptools/pull/3064#discussion_r803011890.)